snapshot/snapshotfs/upload.go

```diff
@@ -744,6 +744,11 @@ func (u *Uploader) maybeIgnoreCachedEntry(ctx context.Context, ent fs.Entry) fs.
 
 func (u *Uploader) effectiveParallelFileReads(pol *policy.Policy) int {
 	p := u.ParallelUploads
+
+	if p > runtime.NumCPU() {
```

Comment:
given that uploads are not entirely CPU bound there may be benefits to allowing p to be greater than the number of cores to allow max number of cores to do hashing while others are waiting for uploads